src/contracts/FixedPriceARM.sol

```diff
@@ -0,0 +1,128 @@
+        internal
+        override
+        returns (uint256 amountOut)
+    {
+        uint256 price;
+        if (inToken == token0) {
+            require(outToken == token1, "ARM: Invalid token");
+            price = traderate0;
+        } else if (inToken == token1) {
+            require(outToken == token0, "ARM: Invalid token");
+            price = traderate1;
+        } else {
+            revert("ARM: Invalid token");
+        }
+        amountOut = amountIn * price / 1e36;
+
+        // Transfer the input tokens from the caller to this ARM contract
+        inToken.transferFrom(msg.sender, address(this), amountIn);
+
+        // Transfer the output tokens to the recipient
+        uint256 transferAmountOut = _calcTransferAmount(address(outToken), amountOut);
+        outToken.transfer(to, transferAmountOut);
+    }
+
+    function _swapTokensForExactTokens(IERC20 inToken, IERC20 outToken, uint256 amountOut, address to)
+        internal
+        override
+        returns (uint256 amountIn)
+    {
+        uint256 price;
+        if (inToken == token0) {
+            require(outToken == token1, "ARM: Invalid token");
+            price = traderate0;
+        } else if (inToken == token1) {
+            require(outToken == token0, "ARM: Invalid token");
+            price = traderate1;
+        } else {
+            revert("ARM: Invalid token");
+        }
+        amountIn = ((amountOut * 1e36) / price) + 1; // +1 to always round in our favor
+
+        // Transfer the input tokens from the caller to this ARM contract
+        inToken.transferFrom(msg.sender, address(this), amountIn);
+
+        // Transfer the output tokens to the recipient
+        uint256 transferAmountOut = _calcTransferAmount(address(outToken), amountOut);
+        outToken.transfer(to, transferAmountOut);
+    }
+
+    /**
+     * @notice Calculate transfer amount for outToken.
+     * Some tokens like stETH transfer less than the requested amount due to internal mechanics.
+     */
+    function _calcTransferAmount(address, uint256 amount) internal view virtual returns (uint256 transferAmount) {
+        transferAmount = amount;
+    }
+
+    /**
+     * @notice Set exchange rates from an operator account
+     * @param buyT1 The buy price of Token 1 (t0 -> t1), denominated in Token 0. 1e36
+     * @param sellT1 The sell price of Token 1 (t1 -> t0), denominated in Token 0. 1e36
+     */
+    function setPrices(uint256 buyT1, uint256 sellT1) external onlyOperatorOrOwner {
+        uint256 _traderate0 = 1e72 / sellT1; // base (t0) -> token (t1)
+        uint256 _traderate1 = buyT1; // token (t1) -> base (t0)
+        // Limit funds and loss when called by operator
+        if (msg.sender == operator) {
+            uint256 currentFunds = token0.balanceOf(address(this)) + token1.balanceOf(address(this));
+            require(currentFunds > minimumFunds, "ARM: Too much loss");
```

Comment:
We could remove this minimum funds stuff - it's not going to make sense in an LP world. (and we get the price defense from locking in the high side at 1 or above.

Reply:
ok, I'll remove